src/main/java/seedu/address/logic/parser/FindTagCommandParser.java

```diff
@@ -0,0 +1,27 @@
+package seedu.address.logic.parser;
+
+import seedu.address.logic.Messages;
+import seedu.address.logic.commands.FindTagCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+
+/**
+ * Parses input arguments and creates a new FindTagCommand object.
+ */
+public class FindTagCommandParser implements Parser<FindTagCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments.
+     * @param args the input arguments
+     * @return a FindTagCommand object containing the input string
+     * @throws ParseException if the user input is empty.
+     */
+    public FindTagCommand parse(String args) throws ParseException {
+        if (args.isEmpty() || args.equals(" ")) {
```

Comment:
second condition would be better as args.trim().equals("") in case of multiple whitespaces.

Reply:
My intent is for the parser to prevent inputs that don't give a keyword, so this part detects whether the user enters "findtag" or "findtag " (with space), but will allow input like "findtag  " to search " ". I'll write a comment explaining this but idt it needs changing since it achieves the intended purpose.

Reply:
got it